src/components/structures/FilePanel.tsx

```diff
@@ -45,6 +45,7 @@ interface IProps {
     roomId: string;
     onClose: () => void;
     resizeNotifier: ResizeNotifier;
+    context: React.ContextType<typeof RoomContext>;
```

Comment:
I've been dithering about whether I want you to document these, and I've decided that I do, if only to tell callers that they don't need to worry about them.

You could argue that `FilePanel` (as defined on line 58, not as exported by `FilePanel.ts` :exploding_head:) is not a public component, and therefore is not subject to the rules of the coding style doc that say "properties must be documented", but I think that is not in keeping with the spirit of the coding style doc. The point is that if I want to use a `FilePanel`, this is going to be my reference to the properties I need to provide, and I don't want to have to reverse-engineer the `forwardRef` incantation to realise that I don't need to specify a `context` here.

```suggestion

    /** Details of the room in which this file belongs. Populated automatically from the `RoomContext`. */
    context: React.ContextType<typeof RoomContext>;
```

Likewise throughout.

This also ties into making the `mxClient` properties explicit rather than inheriting from `MatrixClientProps`: it gives a place to document the fact that callers don't need to populate `mxClient` provided there is a `MatrixClientContext`.